flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/ZooKeeperDefaultDispatcherRunnerTest.java

```diff
@@ -144,8 +147,9 @@ public void teardown() throws Exception {
 
     /** See FLINK-11665. */
     @Test
-    public void testResourceCleanupUnderLeadershipChange() throws Exception {
-        final TestingRpcService rpcService = testingRpcServiceResource.getTestingRpcService();
+    void testResourceCleanupUnderLeadershipChange() throws Exception {
+        final TestingRpcService rpcService =
+                testingRpcServiceExtensionWrapper.getCustomExtension().getTestingRpcService();
         final TestingLeaderElectionService dispatcherLeaderElectionService =
                 new TestingLeaderElectionService();
 
```

Comment:
should we also put `ZooKeeperUtils.startCuratorFramework` which is one line below to `try` resource?